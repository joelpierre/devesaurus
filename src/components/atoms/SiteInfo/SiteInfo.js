import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { graphql, StaticQuery } from 'gatsby';

import styles from './SiteInfo.module.scss';

export const PureSiteInfo = ({ allWordpressSiteMetadata, classes }) => (
  <div
    data-test="component-site-info"
    className={classNames([classes, styles['site-info']])}
  >
    <span className={styles['site-info__heading']}>
      {allWordpressSiteMetadata.edges[0].node.name}
    </span>
    <span className={styles['site-info__slogan']}>
      {allWordpressSiteMetadata.edges[0].node.description}
    </span>
  </div>
);

const SiteInfo = ({ classes }) => (
  <StaticQuery
    data-test="component-site-info-query"
    query={graphql`
      {
        allWordpressSiteMetadata {
          edges {
            node {
              name
              description
            }
          }
        }
      }
    `}
    render={props => <PureSiteInfo {...props} classes={classes} />}
  />
);

SiteInfo.defaultProps = {
  classes: null,
};

SiteInfo.propTypes = {
  classes: PropTypes.string,
};

export default SiteInfo;
