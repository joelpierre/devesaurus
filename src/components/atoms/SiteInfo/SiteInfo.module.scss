@import "~sass/core";

.site-info {
  $root: &;
  @include sr-only;

  &__heading {
    display: block;
  }

  &__slogan {
    display: block;
  }
}
