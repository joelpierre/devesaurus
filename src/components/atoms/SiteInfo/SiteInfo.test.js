import React from 'react';
import { shallow } from 'enzyme';
import { StaticQuery } from 'gatsby';
import toJson from 'enzyme-to-json';

import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../../../utils/test-utilities';

import SiteInfo, { PureSiteInfo } from './SiteInfo';

const defaultProps = {};

const defaultPureProps = {
  allWordpressSiteMetadata: {
    edges: [
      {
        node: {
          name: 'test name',
          description: 'test description',
        },
      },
    ],
  },
};

/**
 * Factory function to create a ShallowWrapper for the SiteInfo component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<SiteInfo {...setupProps} />);
};

const pureSetup = (props = {}) => {
  const setupProps = { ...defaultPureProps, ...props };
  return shallow(<PureSiteInfo {...setupProps} />);
};

describe('<SiteInfo/>', () => {
  let wrapper;
  let pureWrapper;

  beforeEach(() => {
    wrapper = setup();
    pureWrapper = pureSetup();
  });

  it('renders the SiteInfo Component without errors query', () => {
    const component = findByTestAttr(wrapper, 'component-site-info-query');
    expect(component.length).toBe(1);
  });

  it('renders the SiteInfo Component without errors info', () => {
    const component = findByTestAttr(pureWrapper, 'component-site-info');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(SiteInfo, defaultProps);
  });

  it('should render with correct props', () => {
    checkProps(PureSiteInfo, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });

  it('should match snapshot', () => {
    matchSnapshot(pureWrapper);
  });
});
