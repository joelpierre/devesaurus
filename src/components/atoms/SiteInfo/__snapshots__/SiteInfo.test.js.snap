// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteInfo/> should match snapshot 1`] = `
<mockConstructor
  render={[Function]}
/>
`;

exports[`<SiteInfo/> should match snapshot 2`] = `
<div
  className="site-info"
  data-test="component-site-info"
>
  <span
    className="site-info__heading"
  >
    test name
  </span>
  <span
    className="site-info__slogan"
  >
    test description
  </span>
</div>
`;
